packages/react/src/components/Tooltip/index.tsx

```diff
@@ -143,35 +138,35 @@ export default function Tooltip({
     return () => {
       targetElement.removeEventListener('keyup', handleEscape);
     };
-  }, [showProp]);
+  }, [showTooltip]);
 
+  // Handle hover and focus events for the targetElement
   useEffect(() => {
-    if (typeof showProp !== undefined) {
-      targetElement?.addEventListener('mouseenter', handleTriggerMouseEnter);
-      targetElement?.addEventListener('mouseleave', handleTriggerMouseLeave);
-      targetElement?.addEventListener('focusin', show);
-      targetElement?.addEventListener('focusout', hide);
-    }
+    targetElement?.addEventListener('mouseenter', show);
+    targetElement?.addEventListener('mouseleave', hide);
+    targetElement?.addEventListener('focusin', show);
+    targetElement?.addEventListener('focusout', hide);
+
     return () => {
       targetElement?.removeEventListener('mouseenter', show);
       targetElement?.removeEventListener('mouseleave', hide);
```

Comment:
We actually had a mismatch here with "adding" event listeners for `handleTriggerMouseEnter` and `handleTriggerMouseLeave` but then removing `hide` and `show`. Hopefully, now that there is only `hide` and `show`, it will make it easier to do the right thing.

I considered trying to couple the add and remove event listeners more, but I struggled to come up with something that wouldn't just suffer from other short comings (plus, this PR has enough changes already :) )